http://developer.pidgin.im/viewmtn/revision/info/8ae38e4108c4c7c302ab389b0491db4a884bad40
Don't populate the password field on the modify account dialog if we are
not actually remembering it permanently

diff -urN pidgin-2.5.2.orig/pidgin/gtkaccount.c pidgin-2.5.2/pidgin/gtkaccount.c
--- pidgin-2.5.2.orig/pidgin/gtkaccount.c	2008-10-18 12:51:50.000000000 -0400
+++ pidgin-2.5.2/pidgin/gtkaccount.c	2008-11-21 13:44:19.000000000 -0500
@@ -562,7 +562,8 @@
 
 	/* Set the fields. */
 	if (dialog->account != NULL) {
-		if (purple_account_get_password(dialog->account))
+		if (purple_account_get_password(dialog->account) &&
+		    purple_account_get_remember_password(dialog->account))
 			gtk_entry_set_text(GTK_ENTRY(dialog->password_entry),
 							   purple_account_get_password(dialog->account));
 
